avr-hal-generic/src/wdt.rs

```diff
@@ -0,0 +1,126 @@
+//! WDT Implementation
+
+
+pub use embedded_hal::watchdog;
+
+/// Implement traits for the watchdog interface
+#[macro_export]
+macro_rules! impl_wdt {
+    (
+        pub enum WatchdogTimeOutPeriod $supported_periods:tt
+
+        $(#[$wdt_attr:meta])*
+        pub struct $Wdt:ident {
+            mcu_status_register: $MCUSR:ty,
+            peripheral: $WDT:ty,
+            prescaler_bits: {$($period:path => $($bits:ident()).+ ),+}
+        }
+    ) => {
+        /// Approximate length of the time-out period before the watchdog provides a system reset.
+        ///
+        /// After enabling the watchdog timer, call [`Watchdog::feed`] before the period ends to prevent a
+        /// reset.
+        ///
+        /// [`Watchdog::feed`]: watchdog/trait.Watchdog.html#tymethod.feed
+        #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+        pub enum WatchdogTimeOutPeriod $supported_periods
```

Comment:
One minor nit:  I suggest naming the enum just `Timeout` to keep it short.  Otherwise code referring to this type will be streched across the screen quite a bit for no real benefit.